Dockerfile

```diff
@@ -15,54 +8,47 @@ WORKDIR /app
 COPY package.json package-lock.json* ./
 RUN npm install prisma --omit=dev
 
-FROM prisma_base AS deps
+FROM prisma_base AS builder
+ENV NEXT_TELEMETRY_DISABLED=1
 WORKDIR /app
 
 # Won't need to re-install prisma, since it's already installed in the prisma_base image
 RUN npm ci
-
-FROM base AS builder
-WORKDIR /app
-COPY --from=deps /app/node_modules ./node_modules
 COPY . .
+RUN npx prisma generate
+RUN npm run build
 
-# Disable telemetry during the build.
-ENV NEXT_TELEMETRY_DISABLED 1
+FROM base AS runner
+ENV NEXT_TELEMETRY_DISABLED=1
 
-RUN npm run build
+# Make UPLOAD_DIRs
+ENV UPLOAD_DIR=/uploads
+RUN mkdir $UPLOAD_DIR
+RUN mkdir $UPLOAD_DIR/banners $UPLOAD_DIR/cvs $UPLOAD_DIR/avatars $UPLOAD_DIR/logos $UPLOAD_DIR/attachments
 
-# Production image, copy all the files and run next
-FROM app_base AS runner
 WORKDIR /app
 
-# Disable telemetry during runtime.
-ENV NEXT_TELEMETRY_DISABLED 1
+# Take ownership of the app folder and uploads
+RUN chown node:node /app
+RUN chown -R node:node $UPLOAD_DIR
 
-COPY --from=builder /app/public ./public
 
 # Set the correct permission for prerender cache
 RUN mkdir .next
 RUN chown node:node .next
 
-# Copy prisma schema
-COPY --from=builder --chown=node:node /app/prisma ./prisma
-
-# Copy over the prisma client
-COPY --from=prisma_base /app/node_modules ./node_modules
-
-# Automatically leverage output traces to reduce image size
-# https://nextjs.org/docs/advanced-features/output-file-tracing
+COPY --from=builder /app/public ./public
 COPY --from=builder --chown=node:node /app/.next/standalone ./
 COPY --from=builder --chown=node:node /app/.next/static ./.next/static
-
-# Take ownership of the app folder and uploads
-RUN chown node:node /app
-RUN chown -R node:node $UPLOAD_DIR
+COPY --from=builder --chown=node:node /app/prisma ./prisma
+COPY --from=prisma_base /app/node_modules ./node_modules
```

Comment:
Do you need a `prisma builder` and `prisma base` at all here?
`prisma` is in your dependencies, so when you run `npm run ci` you are installing that too already :)